File: socket-app-python/client/basic_client.py
import socket
import base64
import sys

class BasicClient:
    
    def __init__(self, server_address, server_port):
        self.server_address = server_address
        self.server_port = server_port

    def start(self):
        try:
            with socket.create_connection((self.server_address, self.server_port)) as sock:
                sock_stream = sock.makefile(mode='rwb')

                print("Enter your username: ", end='')
                sys.stdout.flush()
                user_name = input()
                sock_stream.write(user_name.encode('utf-8'))
                sock_stream.flush()

                print("Connected to server. Type 'quit' to exit.")
                while True:
                    print("Enter message: ", end='')
                    sys.stdout.flush()
                    user_input = input()

                    if user_input.lower() == 'quit':
                        break

                    message_bytes = user_input.encode('utf-8')
                    sock.sendall(message_bytes)

        except Exception as e:
            print(f"Client exception: {e}")

if __name__ == "__main__":
    port = 12347

    if len(sys.argv) > 1:
        port = int(sys.argv[1])

    client = BasicClient(server_address='localhost', server_port=port)
    client.start()

File: socket-app/src/main/java/com/socket/client/BasicClient.java
package com.socket.client;

import java.io.BufferedOutputStream;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.Socket;

public class BasicClient {

    private String serverAddress;
    private Integer serverPort;

    public BasicClient(String serverAddress, Integer serverPort) {
        this.serverAddress = serverAddress;
        this.serverPort = serverPort;
    }

    public void start() {
        try (Socket socket = new Socket(serverAddress, serverPort);
             BufferedOutputStream bos = new BufferedOutputStream(socket.getOutputStream());
             BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
             BufferedReader stdIn = new BufferedReader(new InputStreamReader(System.in))) {

            System.out.print("Enter your username: ");
            String userName = stdIn.readLine();
            bos.write(userName.getBytes());
            bos.flush();

            System.out.println("Connected to server. Type 'quit' to exit.");
            String userInput;

            while (true) {
                System.out.print("Enter message: ");
                userInput = stdIn.readLine();

                if ("quit".equalsIgnoreCase(userInput)) {
                    break;
                }

                byte[] messageBytes = userInput.getBytes();
                int offset = 0;
                while (offset < messageBytes.length) {
                    int chunkSize = Math.min(messageBytes.length - offset, 2048);
                    bos.write(messageBytes, offset, chunkSize);
                    bos.flush();
                    offset += chunkSize;
                }
            }
        } catch (IOException e) {
            System.err.println("Client exception: " + e.getMessage());
            e.printStackTrace();
        }
    }
}
